From 40b07ad2fcf7d4e64207b6fea60ff44b490ad470 Mon Sep 17 00:00:00 2001
Message-Id: <40b07ad2fcf7d4e64207b6fea60ff44b490ad470.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Thu, 24 Jul 2014 19:04:03 +0530
Subject: [PATCH 250/312] FOR_UPSTREAM [VPG]: drm/i915: Mask
 GEN6_PM_RP_DOWN_TIMEOUT for gen8

Based on the Spec from gen8 onwards, RP down timeout Interrupt is not
supported for Display Interface.

Issue: GMIN-3329
Change-Id: Ie715f27bebd5227e578a7234d1053d1668b10736
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c |    9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index dff5e65..086297c 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -3519,8 +3519,13 @@ static u32 gen6_rps_pm_mask(struct drm_i915_private *dev_priv, u8 val)
 {
 	u32 mask = 0;
 
-	if (val > dev_priv->rps.min_freq_softlimit)
-		mask |= GEN6_PM_RP_DOWN_THRESHOLD | GEN6_PM_RP_DOWN_TIMEOUT;
+	if (val > dev_priv->rps.min_freq_softlimit) {
+		mask |= GEN6_PM_RP_DOWN_THRESHOLD;
+
+		if (INTEL_INFO(dev_priv->dev)->gen <= 7)
+			mask |= GEN6_PM_RP_DOWN_TIMEOUT;
+	}
+
 	if (val < dev_priv->rps.max_freq_softlimit)
 		mask |= GEN6_PM_RP_UP_THRESHOLD;
 
-- 
1.7.9.5

